### ISSUE TITLE
SA1205 - wrong fix with private nested partial class

### ISSUE THREAD

**Issue by kindermannhubert** [DotNetAnalyzers/StyleCopAnalyzers#2114]
Fix for SA1205 does not work correctly with private nested partial classes (for internal and public ones works correctly).
When you have code:
```csharp
    internal partial class C1
    {
        private partial class C2
        {
        }

        partial class C2
        {
        }
    }
```
SA1205 diagnostic is detected for "partial class C2". After fixing code will become:
```csharp
    internal partial class C1
    {
        private partial class C2
        {
        }

        internal partial class C2
        {
        }
    }
```
which does not even compile.

**Comment by bjornhellander** [comment]
Ah, I've been looking for a simple issue for a while, for learning purposes. I can have a go at this if you let me.

**Comment by vweijsters** [comment]
@bjornhellander Consider it yours. I can't assign it to you, as you are not marked as collaborator yet. @sharwell needs to do that.
